### PR TITLE
Update `search_params.py`

### PR DESCRIPTION
**Pull request by Eleven1Liu**
**Description**
1. Small refactoring - finish testing on `MIMIC-50/caml_tune.yml` and `rcv1/cnn_tune.yml`
    - Reuse torch trainer in `search_params.py`.
    - Raise error if no validation set is provided during parameter search.
2. Update `CLIReporter` 
    - Replace hot-fix code with `sort_by_metric=True`.
    - ray version: >= 1.5.0

### CODE REVIEW

File: search_params.py [hunk 1]
```diff
@@ -40,85 +39,25 @@ def step(self):
...
-            )
-            test_metric_dict = trainer.test(
-                best_model, test_dataloaders=test_loader)[0]
+            test_metric_dict = trainer.test()
             for k, v in test_metric_dict.items():
                 test_val_results[f'test_{k}'] = v
 
         # return best val result
-        val_metric_dict = trainer.test(best_model, test_dataloaders=val_loader)[0]
+        val_split = 'val' if 'val' in self.datasets else 'train'
```
Comment (Eleven1Liu) [inline search_params.py:59]: Issue warning message here

**Review by cjlin1 — APPROVED** [review]

**Review by henryyang42 — APPROVED** [review]
